tensorflow_serving/batching/batching_session.cc

```diff
@@ -350,8 +369,15 @@ Status BatchingSession::MergeInputTensors(
       const Tensor& tensor = entry.second;
 
       std::vector<Tensor>& tensor_vec = tensors_to_merge[tensor_name];
-      tensor_vec.push_back(tensor);
-
+      Tensor new_tensor = tensor;
```

Comment:
Instead of new_tensors can we do:

Tensor optionally_padded_tensor;
if (options_.pad_variable_length_inputs) {
  ...
  ... AddPadding(..., &optionally_padded_tensor);
} else {
  optionally_padded_tensor = tensor;
}
...